EIPS/eip-2537.md

```diff
@@ -338,13 +338,14 @@ There are no backward compatibility questions.
 
 ### Subgroup checks
 
-A subgroup check **is mandatory** during the pairing call. Implementations *should* use fast subgroup checks: at the time of writing, multiplication gas cost is based on the `double-and-add` multiplication method that has a clear "worst case" (all bits are equal to one). For pairing operations, it is expected that implementations use faster subgroup checks, e.g. by using the wNAF multiplication method for elliptic curves that is ~ `40%` cheaper with windows size equal to 4. (Tested empirically. Savings are due to lower hamming weight of the group order and even lower hamming weight for wNAF. Concretely, subgroup check for both G1 and G2 points in a pair are around `35000` combined).
-
+The check **is mandatory** during **ALL** the calls. Implementations *should* use fast subgroup checks. It can be stated that it **must** be performed by fast subgroup checks to achieve a speedup, resulting in a discount over naive implementations based on the `double-and-add` multiplication method, which has a clear "worst-case" scenario where all bits are equal to one.
+Before accepting an input, it is recommended to subject the input to the appropriate endomorphism test as described in https://eprint.iacr.org/2021/1130.pdf.
```

Comment:
MUST perform a subgroup check, and reject invalid point.

Reply:
@mratsim  what do you mean? is there anything wrong with this part of the text ?

Reply:
AFAIK, in RFC **MUST** is the way to specify that something is mandatory, and the spec should explicitly mention that it returns as an error on failure.

Reply:
@mratsim  Is the problem the capitalization of 'must' rather than using lowercase? Or am I missing something else? :)

Reply:
To be clear, I have 2 issues:

1. Instead of **is mandatory**: "scalar multiplications, MSMs and pairings MUST perform a subgroup check. Implementations SHOULD use the fast subgroup check method from 2021/1130 detailed in _link to fast subgroupcheck markdown file_
2. "On inputs that fail the subgroup check the precompile MUST return an error."

Rationale:
- MUST is how mandatory steps are usually specified. I added details since EC add and hashing-to-curve do not have subgroup checks. And now we have a dedicated document for fast subgroup-checks.
- Similar to the paragraph on _Field elements encoding_ with "On inputs that can not be a valid encodings of field elements the precompile must return an error.", it's important to spell what happens on failure or people have to look at the implementations to make sure. For example what if for MSM, the offending point and its corresponding coefficient were skipped?

I can make a PR to this PR.

Reply:
>I can make a PR to this PR.

@mratsim  this would be great.
One you are on it I everyone agrees you can already to the PR to this PR with this strategy:

2. Misuse resistance approach. See [@asanso](https://ethereum-magicians.org/u/asanso), enforce subgroup check except on addition.

assuming @shamatar @ralexstokes @ineffectualproperty agree